src/Symfony/Component/OptionsResolver/OptionConfigurator.php

```diff
@@ -0,0 +1,130 @@
+<?php
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver;
+
+use Symfony\Component\OptionsResolver\Exception\AccessException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+
+final class OptionConfigurator
+{
+    /**
+     * The name of the defined option
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var OptionsResolver
+     */
+    private $resolver;
+
+    public function __construct(string $name, OptionsResolver $resolver)
+    {
+        $this->name = $name;
+        $this->resolver = $resolver;
+        $this->resolver->setDefined($name);
+    }
+
+    /**
+     * Adds allowed types for this option.
+     *
+     * @param string|string[] $types One or more accepted types
+     *
+     * @return $this
+     *
+     * @throws UndefinedOptionsException If the option is undefined
```

Comment:
should be removed. The option will never be undefined, as the constructor defines it.

Reply:
this applies to all methods